### PR TITLE
webrtc: use union types instead of enums

### PR DESCRIPTION
**Pull request by shaharmor**
case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

---
- use union types instead of `any` for `RTCMessageEvent` `data` property
- replace `RTCSdpType`, `RTCDataChannelState`, `RTCIceGatheringState`, `RTCIceConnectionState`, `RTCSignalingState` enums with string union types.

### CODE REVIEW

File: webrtc/RTCPeerConnection.d.ts [hunk 1]
```diff
@@ -121,33 +118,19 @@ interface RTCDataChannelInit {
...
-  'answer'
-}
+type RTCSdpType = 'offer' | 'pranswer' | 'answer' | 'rollback';
 
 interface RTCMessageEvent {
-  // http://dev.w3.org/2011/webrtc/editor/webrtc.html#event-datachannel-message
-  // At present, this can be an: ArrayBuffer, a string, or a Blob.
-  // See TODO(2)
-  data: any;
+  data: ArrayBuffer | string | Blob;
```
Comment (dbrgn) [inline webrtc/RTCPeerConnection.d.ts]: According to the spec, the `data` has the type `any`. The [Mozilla docs](https://developer.mozilla.org/en-US/docs/Web/API/MessageEvent) write `Returns a DOMString, Blob or an ArrayBuffer containing the data send by the emitter`.

Not sure if it should be changed to `any` or not.

**Comment by dbrgn** [comment]
I reviewed everything in this PR. Besides the comment above, everything looks correct according to the [spec](https://www.w3.org/TR/webrtc/).

File: webrtc/RTCPeerConnection.d.ts (hunk 1 above)
Comment (vvakame) [inline webrtc/RTCPeerConnection.d.ts]: I vote 👍  for `any`.
https://www.w3.org/TR/webmessaging/#the-messageevent-interfaces

**Comment by mhegazy** [comment]
@shaharmor can you please resolve the merge conflicts, and we should be good to go.

**Comment by RyanCavanaugh** [comment]
Hello and thank you for your contribution!

Due to an excessively long queue of pull requests that have become stale, we are "declaring bankruptcy" and closing all PRs opened before May 1, 2016. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

Going forward, we are committing to review or merge all PRs on a regular basis so this bankruptcy will not occur again. We apologize for the incovenience and hope you will continue to contribute to DefinitelyTyped in the future.

Thanks
The TypeScript and DefinitelyTyped teams

**Pull request closed without merging**